exercises/all-your-base/all_your_base_test.go

```diff
@@ -0,0 +1,197 @@
+package allyourbase
+
+import "testing"
+
+const targetTestVersion = 1
+
+func TestTestVersion(t *testing.T) {
+	if testVersion != targetTestVersion {
+		t.Errorf("Found testVersion = %v, want %v.", testVersion, targetTestVersion)
+	}
+}
+
+var testCases = []struct {
+	description  string
+	inputBase    int64
+	outputBase   int64
+	inputDigits  []int64
+	outputDigits []int64
+}{
+	{
+		description:  "single bit one to decimal",
+		inputBase:    2,
+		inputDigits:  []int64{1},
+		outputBase:   10,
+		outputDigits: []int64{1},
+	},
+	{
+		description:  "binary to single decimal",
+		inputBase:    2,
+		inputDigits:  []int64{1, 0, 1},
+		outputBase:   10,
+		outputDigits: []int64{5},
+	},
+	{
+		description:  "single decimal to binary",
+		inputBase:    10,
+		inputDigits:  []int64{5},
+		outputBase:   2,
+		outputDigits: []int64{1, 0, 1},
+	},
+	{
+		description:  "binary to multiple decimal",
+		inputBase:    2,
+		inputDigits:  []int64{1, 0, 1, 0, 1, 0},
+		outputBase:   10,
+		outputDigits: []int64{4, 2},
+	},
+	{
+		description:  "decimal to binary",
+		inputBase:    10,
+		inputDigits:  []int64{4, 2},
+		outputBase:   2,
+		outputDigits: []int64{1, 0, 1, 0, 1, 0},
+	},
+	{
+		description:  "trinary to hexadecimal",
+		inputBase:    3,
+		inputDigits:  []int64{1, 1, 2, 0},
+		outputBase:   16,
+		outputDigits: []int64{2, 10},
+	},
+	{
+		description:  "hexadecimal to trinary",
+		inputBase:    16,
+		inputDigits:  []int64{2, 10},
+		outputBase:   3,
+		outputDigits: []int64{1, 1, 2, 0},
+	},
+	{
+		description:  "15-bit int64eger",
+		inputBase:    97,
+		inputDigits:  []int64{3, 46, 60},
+		outputBase:   73,
+		outputDigits: []int64{6, 10, 45},
+	},
+	{
+		description:  "empty list",
+		inputBase:    2,
+		inputDigits:  []int64{},
+		outputBase:   10,
+		outputDigits: nil,
```

Comment:
Does one way mathematically make more sense? Is one way likely lead to more elegant solutions?

to me, empty input list of digits implies that the value is zero. I might suggest that the output be `[]int64{0}`